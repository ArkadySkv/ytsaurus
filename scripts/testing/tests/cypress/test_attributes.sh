#!/bin/bash
#% NUM_MASTERS = 3
#% NUM_HOLDERS = 0

yt set //root '{nodes=[1; 2]} <attr=100;mode=rw>'
yt get //root@
yt get //root@attr

yt remove //root@
yt get //root@

yt remove //root/nodes
yt get //root

echo 'changing attributes'
yt set //root/2 '[] < author=ignat >'
yt get //root/2
yt get //root/2@
yt get //root/2@author

yt set //root/2@author not_ignat
yt get //root/2@author 

echo 'nested attributes'
yt set //root/3 '[] <dir=<file=-100<>>>' 
yt get //root/3@
yt get //root/3@dir@
yt get //root/3@dir@file
yt get //root/3@dir@file@
