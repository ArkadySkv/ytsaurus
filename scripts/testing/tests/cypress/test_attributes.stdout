{"size"=1;"parent_id"="ffffffff-ffffffff-12f-140a8383";"lock_mode"="none";"lock_ids"=[];"subtree_lock_ids"=[];"id"="0-6-12f-f705969";"type"="map_node";"ref_counter"=1;"attr"=100;"mode"="rw"}
100
{"size"=1;"parent_id"="ffffffff-ffffffff-12f-140a8383";"lock_mode"="none";"lock_ids"=[];"subtree_lock_ids"=[];"id"="0-6-12f-f705969";"type"="map_node";"ref_counter"=1}
{}
changing attributes
[]
{"size"=0;"parent_id"="0-6-12f-f705969";"lock_mode"="none";"lock_ids"=[];"subtree_lock_ids"=[];"id"="0-2-130-6cadc45e";"type"="list_node";"ref_counter"=1;"author"="ignat"}
"ignat"
"not_ignat"
nested attributes
{"size"=0;"parent_id"="0-6-12f-f705969";"lock_mode"="none";"lock_ids"=[];"subtree_lock_ids"=[];"id"="0-3-130-9d7bb2ee";"type"="list_node";"ref_counter"=1;"dir"=<"file"=-100<>>}
{"file"=-100<>}
-100<>
{}
